sdk/src/worker.ts

```diff
@@ -0,0 +1,145 @@
+import { initializeWasm, initThreadPool, ProgramManager, PrivateKey, verifyFunctionExecution } from "./browser";
+import { AleoKeyProvider, AleoKeyProviderParams} from "./function-key-provider";
+import { expose } from "comlink";
+
+await initializeWasm();
+await initThreadPool(10);
+
+const defaultHost = "https://vm.aleo.org/api";
+const keyProvider = new AleoKeyProvider();
+const programManager = new ProgramManager(
+    defaultHost,
+    keyProvider,
+    undefined
+);
+
+keyProvider.useCache(true);
+
+// self.postMessage({
+//     type: "ALEO_WORKER_READY",
+// });
+
+export interface WorkerAPI {
+    additionalFunction1: () => Promise<string>;
+    additionalFunction2: (data: any) => Promise<any>;
+}
+
+let lastLocalProgram: string = "";
+
+export interface WorkerAPI {
+    executeOffline: (
+        localProgram: string,
+        aleoFunction: string,
+        inputs: string[],
+        privateKey: string
+    ) => Promise<{ outputs: any; execution: string } | string>;
+
+    getPrivateKey: () => Promise<PrivateKey>;
+}
+async function executeOffline(
+    localProgram: string,
+    aleoFunction: string,
+    inputs: string[],
+    privateKey: string
+) {
+    console.log("Web worker: Executing function locally...");
+    let startTime = performance.now();
+
+    try {
+        // Ensure the program is valid and that it contains the function specified
+        const program = programManager.createProgramFromSource(localProgram);
+        if (program instanceof Error) {
+            throw "Error creating program from source";
+        }
+        const program_id = program.id();
+        if (!program.hasFunction(aleoFunction)) {
+            throw `Program ${program_id} does not contain function ${aleoFunction}`;
+        }
+        const cacheKey = `${program_id}:${aleoFunction}`;
+
+        // Get the program imports
+        const imports = await programManager.networkClient.getProgramImports(
+            localProgram
+        );
+
+        if (imports instanceof Error) {
+            throw "Error getting program imports";
+        }
+        // Get the proving and verifying keys for the function
+        if (lastLocalProgram !== localProgram) {
+            const keys = programManager.executionEngine.synthesizeKeypair(
+                localProgram,
+                aleoFunction
+            );
+            programManager.keyProvider.cacheKeys(cacheKey, [
+                keys.provingKey(),
+                keys.verifyingKey(),
+            ]);
+            lastLocalProgram = localProgram;
+        }
+
+        // Pass the cache key to the execute function
+        const keyParams = new AleoKeyProviderParams({
+            cacheKey: cacheKey,
+        });
+
+        // Execute the function locally
+        let response = await programManager.executeOffline(
+            localProgram,
+            aleoFunction,
+            inputs,
+            false,
```

Comment:
This should just be changed to proveExecution to provide the option to use it.

```suggestion
            proveExecution,
```